bucket-policy.go

```diff
@@ -390,3 +391,99 @@ func unMarshalBucketPolicy(bucketPolicyBuf []byte) (BucketAccessPolicy, error) {
 	}
 	return policy, nil
 }
+
+// Identifies the policy type from policy Statements.
+func identifyPolicyType(policy BucketAccessPolicy, bucketName, objectPrefix string) (bucketPolicy BucketPolicy) {
+	if policy.Statements == nil {
+		return BucketPolicyNone
+	}
+	if isBucketPolicyReadWrite(policy.Statements, bucketName, objectPrefix) {
+		return BucketPolicyReadWrite
+	} else if isBucketPolicyWriteOnly(policy.Statements, bucketName, objectPrefix) {
+		return BucketPolicyWriteOnly
+	} else if isBucketPolicyReadOnly(policy.Statements, bucketName, objectPrefix) {
+		return BucketPolicyReadOnly
+	}
+	return BucketPolicyNone
+}
+
+// Generate policy statements for variosu bucket polices.
+// Fills in the statement structure of the parameter `policy`, and returns the modified policy.
```

Comment:
--> 'Fills in the statement structure for the parameter `policy`, and returns the modified policy.